iFixit/react-commerce
# Remove featured product list section from product list page - frontend changes

closes #1483

This PR removes the featured-product-list section since the new mockups do not include that component anymore.

Since the featured-product-list section was relying on Algolia, this forced us to put in place a ["multi-index" Algolia configuration](https://www.algolia.com/doc/guides/building-search-ui/ui-and-ux-patterns/multi-index-search/react-hooks/#searching-the-same-index-with-different-search-parameters) for the page, and this seems to bring some unwanted side effects. So this is also an opportunity to get rid of that. 

In this PR we remove **all the frontend code** linked to the featured-product-list component.
Another PR will follow for the Strapi cleanup.

<img width="1343" alt="Screenshot 2023-03-20 at 17 48 16" src="https://user-images.githubusercontent.com/7805759/226582735-2828af62-bb5c-45d0-9135-7fd86fcae3a5.png">

### Detailed explanation

Having multiple indexes on the page means that the internal Algolia state needs to take into account both a facet-filtered search (the one associated to the main content) and a search without any facet (the one associated to a featured-product-list section).
So we effectively perform two queries towards Algolia.

By inspecting how the Algolia state updates, there is a short time window in which the internal Algolia state clears the list of facets. This is what causes the flicker described in #1483.
This was present also before we chose to hide the facets panel if empty: it was only less visible because the panel remained in place but the facets were briefly removed and then re-added.
Please note that this is not a transition from the previous facets to the new facets since the new facets were already visible before the facets were temporarily emptied.

You can see the two queries, the old behaviour of facets and all the details described above in the following video:

https://user-images.githubusercontent.com/7805759/226594651-1d1badc8-a2cd-4da2-b8ae-6fccf6e5cd92.mov

### QA

1. Visit Vercel preview
2. Verify that product list pages are still displaying correctly
3. Verify that the issue described in #1483 is not present anymore
4. Verify that we are now performing a single query towards Algolia (while performing client side navigation)
